kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -130,6 +134,41 @@ public static UnsupportedOperationException unsupportedWriteFeature(
                 writeProtocolVersion, writeFeature));
     }
 
+    public static IllegalArgumentException unsupportedDataType(DataType dataType) {
+        return new IllegalArgumentException("Unsupported data type: " + dataType);
+    }
+
+    public static IllegalArgumentException unsupportedPartitionDataType(
+            String colName,
+            DataType dataType) {
+        String msg = format("Unsupported partition column (%s) with type: %s", colName, dataType);
+        return new IllegalArgumentException(msg);
+    }
+
+    public static IllegalArgumentException duplicateColumnsInSchema(
+            StructType schema,
+            List<String> duplicateColumns) {
+        String msg = format(
+            "Schema contains duplicate columns: %s.\nSchema: %s",
+            String.join(", ", duplicateColumns),
+            schema);
+        return new IllegalArgumentException(msg);
+    }
+
+    public static IllegalArgumentException invalidColumnName(
+            String columnName,
+            String unsupportedChars) {
+        return new IllegalArgumentException(format(
+                "Attribute name '%s' contains one of the unsupported (%s) characters.",
+                columnName,
+                unsupportedChars));
+    }
+
+    public static IllegalArgumentException requiresSchemaForNewTable(String tablePath) {
```

Comment:
Can we make this more user facing? Can this just be a general "table not found - cannot write to a non-existent table" exception? Needing to provide a schema should be more connector-facing

Reply:
Hmm or should this just be a `TableNotFound` exception?